cache-filter/src/filter/http.rs

```diff
@@ -292,7 +297,7 @@ impl Context for CacheFilter {
         match self.get_http_call_response_body(0, body_size) {
             Some(bytes) => {
                 match Authorization::from_str(std::str::from_utf8(&bytes).unwrap()) {
-                    Ok(Authorization::Ok(response)) => {
+                    Ok(Authorization::Status(response)) => {
                         if let Err(e) = self.handle_auth_response(&response) {
```

Comment:
Note that you are passing in the `AuthorizationStatus` and `handle_auth_response` just goes on to pick the usage reports, but does not actually check that the authorization is accepted/authorized or rejected/denied.

Reply:
@unleashed This is actually on the cache filter side. I didn't change anything related to the implementation. Just changed the Authorization to use the new API. eg: `Authorization::Ok`  to `Authorization::Status`. And yes I think `handle_auth_response` should take account status as well. We will get both 200 and 409 through that `Authorization::Ok`. `handle_auth_response` in singleton service takes status into account. https://github.com/NomadXD/gsoc-wasm-filters/blob/dev/auth-call/singleton-service/src/service/proxy.rs#L252
Will fix this in a separate PR. cc @rahulanand16nov

Reply:
@NomadXD Better do it in this PR as it's a small change - let me know if you need any help with it.

Reply:
fixed